### ISSUE TITLE
Releases are not installed

### ISSUE THREAD

**Issue by webyneter** [hashicorp/terraform-provider-helm#431]
`stable/nginx-ingress` and `jetstack/cert-manager` end up not being installed when using `helm` terraform provider.

### Terraform Version

```bash
Terraform v0.12.21
+ provider.google v3.10.0
+ provider.google-beta v3.10.0
+ provider.helm v1.0.0
+ provider.kubernetes v1.11.1
+ provider.null v2.1.2
+ provider.random v2.2.1
```

### Affected Resource(s)

- helm_release

### Terraform Configuration Files
```hcl
#### providers.tf

provider "random" {
  version = "~> 2.2"
}

provider "null" {
  version = "~> 2.1"
}

provider "google" {
  # https://www.terraform.io/docs/providers/google/guides/provider_reference.html
  version = "~> 3.10"
  project = var.project
  region  = var.region
  # Expose through CI envs:
  # https://www.terraform.io/docs/providers/google/guides/provider_reference.html#credentials-1
  credentials = file(var.service_account_credentials_file_path)
}

provider "google-beta" {
  # https://www.terraform.io/docs/providers/google/guides/provider_reference.html
  version = "~> 3.10"
  project = var.project
  region  = var.region
  # Expose through CI envs:
  # https://www.terraform.io/docs/providers/google/guides/provider_reference.html#credentials-1
  credentials = file(var.service_account_credentials_file_path)
}

provider "helm" {
  version = "~> 1.0"
}

provider "kubernetes" {
  version = "~> 1.11"
}


#### main.tf

# ...

resource "kubernetes_namespace" "nginx_ingress" {
  metadata {
    name = "nginx-ingress"
  }
}
resource "kubernetes_namespace" "cert_manager" {
  metadata {
    name = "cert-manager"
  }
}

resource "random_id" "nginx_ingress_name_suffix" {
  # https://www.terraform.io/docs/providers/random/r/id.html
  byte_length = local.random_id_byte_length
}
locals {
  nginx_ingress_name    = "${var.environment}-nginx-ingress-${random_id.nginx_ingress_name_suffix.hex}"
  nginx_ingress_version = "1.33.0"
}
resource "helm_release" "nginx_ingress" {
  # https://www.terraform.io/docs/providers/helm/r/release.html
  name       = local.nginx_ingress_name
  repository = data.helm_repository.stable.url
  chart      = "nginx-ingress"
  # region https://github.com/kubernetes/ingress-nginx/releases
  version = local.nginx_ingress_version
  # endregion
  namespace = kubernetes_namespace.nginx_ingress.metadata[0].name
}

resource "random_id" "cert_manager_name_suffix" {
  # https://www.terraform.io/docs/providers/random/r/id.html
  byte_length = local.random_id_byte_length
}
locals {
  cert_manager_name    = "${var.environment}-cert-manager-${random_id.cert_manager_name_suffix.hex}"
  cert_manager_version = "0.13.1"
}
resource "helm_release" "cert_manager" {
  # https://www.terraform.io/docs/providers/helm/r/release.html
  name       = "${var.environment}-cert-manager-${random_id.cert_manager_name_suffix.hex}"
  repository = data.helm_repository.jetstack.url
  chart      = "cert-manager"
  # region https://github.com/jetstack/cert-manager/releases
  version = "0.13.1"
  # endregion
    namespace = kubernetes_namespace.cert_manager.metadata[0].name
}
```

### Debug Output

`terraform plan -out ./tfp`: https://gist.github.com/webyneter/ac75187ee6a75a03de0b46a29db52eed
`terraform apply ./tfp`: https://gist.github.com/webyneter/187d98056dbada3d4b55d93163b8a657

However, querying through kubectl by namespace yields no running pods, same picture when inspecting GCP: `kubectl get pods --namespace nginx-ingress` / `kubectl get pods --namespace cert-manager`.

### Expected Behavior

Chart releases should have been deployed to GCP.

### Actual Behavior

No charts are actually being deployed to GCP.

### Steps to Reproduce

1. `terraform init`
1. `terraform plan`
1. `terraform apply`

### Important Factoids

* GCP
* `kubectl version`: `Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-11T18:14:22Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.10-gke.17", GitCommit:"bdceba0734835c6cb1acbd1c447caf17d8613b44", GitTreeState:"clean", BuildDate:"2020-01-17T23:10:13Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}`
* `helm version`: `version.BuildInfo{Version:"v3.1.1", GitCommit:"afe70585407b420d0097d07b21c47dc511525ac8", GitTreeState:"clean", GoVersion:"go1.13.8"}`

### (Unmanaged) Workaround

```tf
resource "null_resource" "nginx_ingress" {
  provisioner "local-exec" {
    command = "helm install ${local.nginx_ingress_name} --version ${local.nginx_ingress_version} --namespace ${kubernetes_namespace.nginx_ingress.metadata[0].name} stable/nginx-ingress"
  }
}

resource "null_resource" "cert_manager" {
  provisioner "local-exec" {
    command = "helm install ${local.cert_manager_name} --version ${local.cert_manager_version} --namespace ${kubernetes_namespace.cert_manager.metadata[0].name} jetstack/cert-manager"
  }
}
```

**Comment by bishtawi** [comment]
I am seeing this too when attempting to install [prometheus-operator](https://github.com/helm/charts/tree/master/stable/prometheus-operator) (and a few other packages too).

```
resource "helm_release" "prometheus_operator" {
  name       = "prometheus-operator"
  repository = data.helm_repository.stable.metadata.0.name
  chart      = "prometheus-operator"
  namespace  = "monitoring"

  atomic          = true
  cleanup_on_fail = true

  set {
    name  = "rbac.create"
    value = "true"
  }

  set_sensitive {
    name  = "grafana.adminPassword"
    value = random_password.grafana_password.result
  }
}
```

I have the `debug = true` set in the helm provider block but I get no error messages or debug statements in the output to help figure out what is failing. Terraform instead returns a success:
```
module.eks.module.helm.helm_release.prometheus_operator: Creation complete after 1m32s [id=prometheus-operator]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

But when you query the cluster via kubectl you find that nothing was installed.
```
$ kubectl -n monitoring get all

No resources found.
```

I am running:
```
$ terraform version
Terraform v0.12.17
+ provider.auth0 v0.5.2
+ provider.aws v2.53.0
+ provider.helm v1.0.0
+ provider.kubernetes v1.11.1
+ provider.local v1.4.0
+ provider.null v2.1.2
+ provider.random v2.2.1
+ provider.template v2.1.2
+ provider.tls v2.1.1
```

And I am on AWS EKS 1.15.

**Comment by mtverraen** [comment]
It's no solution, but an observation: It seems to have something to do with the "namespace" attribute of the provider.  If I omit "namespace = xyz" in the helm_release, it gets installed just fine into the "default" namespace.

**Comment by bishtawi** [comment]
Yup, did some testing and can confirm that its related to the namespace. Specifically, the namespace needs to exist before you install a package via Helm v3. And for whatever reason, the error is being swallowed.

Workaround is to use `kubernetes_namespace` resources to create the namespace before creating the `helm_release` resource.

Looks like its the same bug as https://github.com/terraform-providers/terraform-provider-helm/issues/427

**Comment by jrhouston** [comment]
This was fixed in 1.1.0.
